flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/TimeWindowPropertyCollector.scala

```diff
@@ -19,34 +19,44 @@
 package org.apache.flink.table.runtime.aggregate
 
 import org.apache.calcite.runtime.SqlFunctions
+import org.apache.flink.table.runtime.types.CRow
 import org.apache.flink.types.Row
 import org.apache.flink.util.Collector
 
 /**
   * Adds TimeWindow properties to specified fields of a row before it emits the row to a wrapped
   * collector.
   */
-class TimeWindowPropertyCollector(windowStartOffset: Option[Int], windowEndOffset: Option[Int])
-    extends Collector[Row] {
+class TimeWindowPropertyCollector[T](windowStartOffset: Option[Int], windowEndOffset: Option[Int])
```

Comment:
we can make this an `abstract` class and add to subclasses `RowTimeWindowPropertyCollector` and `CRowTimeWindowPropertyCollector`.
That would avoid the `if` condition to identify the input type.